### DISCUSSION TITLE
deterministic finite automata

### DISCUSSION THREAD

**Discussion by MrsMaulwurf** [irgroup-classrooms/dis08-2021#35] [Q&A] [answered]
Hallo,
ich habe zur Einheit "Regular Expressions" eine wahrscheinlich sehr banale Frage. Ich weiß aber leider nicht recht, nach was ich suchen soll, um sie mir selbst zu beantworten. Vielleicht stehe ich auch nur auf dem Schlauch.

Bei Folie 5, etwa Minute 16 geht es um den endlichen Automaten.

Von Start S komme ich mit 1 zu A. Wenn dann 0 folgt, woher weiß ich, welchem Pfad jetzt gefolgt wird. Es könnte zu B gehen oder zum Endpunkt. 

Danke im Voraus für die Hilfe!

Viele Grüße
Tina Grahl

**Comment by HaakFab** [comment, marked as answer]
Guten Morgen,

Sie denken Grundsätzlich richtig, müssen sich aber dazu Vorstellen, dass immer "auf einem Input String" gefahren wird. Hat der Input noch weitere Zeichen? Dann geht der Automat zu B. Ist dies nicht der Fall, geht er zum Endpunkt.
Das ist zwar nur die "halbe Erklärung", aber für die Analogie zur RegEx reicht diese Variante. Wenn ich das noch richtig in Erinnerung habe, werden immer alle Pfade bereist, würde noch ein Token kommen könnte dieser vom Endpunkt nicht mehr erreicht werden und der Pfad würde "fallengelassen" werden. 

VG,
Fabian Haak
